Cache/RedisCache.php

```diff
@@ -12,44 +13,55 @@
  */
 class RedisCache extends \Doctrine\Common\Cache\RedisCache
 {
-    /** @var \Redis */
-    private $master;
-    /** @var array of \Redis */
-    private $slaves = [];
+    /** @var array Host, port and timeout for connecting to Redis write server. */
+    private $writeConfig = [];
+    /** @var array[] Host, port and timeout for connecting to Redis write server. */
+    private $readConfigs = [];
+    /** @var Redis */
+    private $write;
+    /** @var Redis */
+    private $read;
+    /** @var LoggerInterface */
+    private $logger;
 
     /**
-     * @param \Redis $redis
+     * @param LoggerInterface|null $logger
      */
-    public function setMaster(Redis $redis)
+    public function __construct(LoggerInterface $logger = null)
     {
-        $redis->setOption(Redis::OPT_SERIALIZER, $this->getSerializerValue());
-        $this->master = $redis;
+        $this->logger = $logger;
     }
 
     /**
-     * @param \Redis $redis
+     * @param Redis $redis
+     * @param $host
+     * @param $port
+     * @param float $timeout
      */
-    public function addSlave(Redis $redis)
+    public function addRead(Redis $redis, $host, $port, $timeout = 0.0)
     {
-        $redis->setOption(Redis::OPT_SERIALIZER, $this->getSerializerValue());
-        $this->slaves[] = $redis;
+        $this->readConfigs[] = [
+            'redis' => $redis,
+            'host' => $host,
+            'port' => $port,
+            'timeout' => $timeout,
+        ];
     }
 
     /**
-     * @return \Redis
-     */
-    public function getMaster()
-    {
-        return $this->master;
-    }
-
-    /**
-     * Retrieve a random slave from the list.
-     * @return \Redis
+     * @param Redis $redis
+     * @param $host
+     * @param $port
+     * @param float $timeout
      */
-    public function getSlave()
+    public function setWrite(Redis $redis, $host, $port, $timeout = 0.0)
```

Comment:
Good question, it's the default value for Redis connect(). 0.0 is treated as "unlimited", so its as if you don't specify a timeout.